src/test/java/com/scaleset/cfbuilder/CloudFormationBuilderTest.java

```diff
@@ -1,73 +1,81 @@
     }
 
     class TestModule extends Module {
+        private static final String KEYNAME_DESCRIPTION = "Name of an existing EC2 KeyPair to enable SSH access to the instances";
+        private static final String KEYNAME_TYPE = "AWS::EC2::KeyPair::KeyName";
+        private static final String KEYNAME_CONSTRAINT_DESCRIPTION = "must be the name of an existing EC2 KeyPair.";
 
         public void build() throws Exception {
+            
+            Parameter keyName = (Parameter) option("KeyName").orElseGet(
+                () -> strParam("KeyName")
+                    .type(KEYNAME_TYPE)
+                    .description(KEYNAME_DESCRIPTION)
+                    .constraintDescription(KEYNAME_CONSTRAINT_DESCRIPTION));
 
-            Object keyName = option("KeyName").orElseGet(
-                    () -> strParam("KeyName").type("AWS::EC2::KeyPair::KeyName").description("Name of an existing EC2 KeyPair to enable SSH access to the instances").constraintDescription("must be the name of an existing EC2 KeyPair."));
-
-            Object clusterName = option("clusterName").orElse("elasticsearch");
             Object cidrIp = "0.0.0.0/0";
             Object keyNameVar = template.ref("KeyName");
-            Object imageId = template.ref("ImageId");
-            Object az = template.ref("MyAZ");
-            Object instanceProfile = ref("InstanceProfile");
-            Object vpcId = template.ref("VpcId");
+            Object webServerSecurityGroupName = template.ref("WebServerSecurityGroup");
+            Object dbEc2SecurityGroupId = template.fnGetAtt("DBEC2SecurityGroup", "GroupId");
 
-            SecurityGroup webServerSecurityGroup = resource(SecurityGroup.class, "WebServerSecurityGroup").groupDescription("Enable ports 80 and 22")
+            SecurityGroup webServerSecurityGroup = resource(SecurityGroup.class, "WebServerSecurityGroup")
+                    .groupDescription("Enable ports 80 and 22")
                     .ingress(ingress -> ingress.cidrIp(cidrIp), "tcp", 80, 22);
-
-            Object groupId = webServerSecurityGroup.fnGetAtt("GroupId");
-
-            Instance webServerInstance = resource(Instance.class, "WebServerInstance").imageId("ami-0def3275")
-                    .instanceType("t2.micro")
-                    .securityGroupIds(webServerSecurityGroup)
-                    .keyName(keyName);
-
-
-            Object value = webServerInstance.fnGetAtt("PublicDnsName");
-
-            output("websiteURL", value, "URL for newly created LAMP stack");
+            SecurityGroup dbEc2SecurityGroup = resource(SecurityGroup.class, "DBEC2SecurityGroup")
+                    .groupDescription("Open database for access")
+                    .ingress(ingress -> ingress.sourceSecurityGroupName(webServerSecurityGroupName), "tcp", 3306);
+
+            Object webServerSecurityGroupId = webServerSecurityGroup.fnGetAtt("GroupId");
+
+            resource(SecurityGroupIngress.class, "SelfReferenceIngress")
```

Comment:
Is this Resource needed? I think not

Reply:
No it's definitely not needed. Must be a remnant of a copy paste. I'll remove it.